internal/provider/resource_silo.go

```diff
@@ -0,0 +1,480 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework-timeouts/resource/timeouts"
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/boolplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/oxidecomputer/oxide.go/oxide"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource              = (*siloResource)(nil)
+	_ resource.ResourceWithConfigure = (*siloResource)(nil)
+)
+
+// NewSiloResource is a helper function to simplify the provider implementation.
+func NewSiloResource() resource.Resource {
+	return &siloResource{}
+}
+
+// siloResource is the resource implementation.
+type siloResource struct {
+	client *oxide.Client
+}
+
+type siloResourceModel struct {
+	AdminGroupName   types.String             `tfsdk:"admin_group_name"`
+	Description      types.String             `tfsdk:"description"`
+	Discoverable     types.Bool               `tfsdk:"discoverable"`
+	ID               types.String             `tfsdk:"id"`
+	IdentityMode     types.String             `tfsdk:"identity_mode"`
+	MappedFleetRoles map[string][]string      `tfsdk:"mapped_fleet_roles"`
+	Name             types.String             `tfsdk:"name"`
+	Quotas           quotaResourceModel       `tfsdk:"quotas"`
```

Comment:
Taking this naming pattern from other files.

```suggestion
	Quotas           siloResourceQuotaModel       `tfsdk:"quotas"`
```